File: ex1.cpp
#include <bits/stdc++.h>
using namespace std;

int greatestRem(int val, vector<int> arr) {
    int n = arr.size();
    int res = INT_MIN;
    int count = 0;

    for (int i = 0; i < n; i++)
    {
        if(arr[i]%val == 0) continue;
        else {
            res = max(res,arr[i]%val);
            count++;
        }
    }

    return res;
    
}

int leastRem(int val, vector<int> arr) {
    int n = arr.size();
    int rem = INT_MAX;
    int count = 0;

    for (int i = 0; i < n; i++)
    {
        if(arr[i]%val == 0) continue;
        else {
            rem = min(rem,arr[i]%val);
            count++;
        }
    }

    if(!count) return 0;
    return rem;
    
}

int main() {
    vector <int> arr = {10,20,30,40,50,44,55,60};
    int ans = greatestRem(10,arr);
    int ans2 = leastRem(10,arr);
    cout << ans << " " << ans2;
    return 0;
}

File: day4/ex_1.py
def func(arr):
    count = {}
    for i in arr:
        if i in count:
            count[i] += 1
        else:
            count[i] = 1
    ans = sorted(count.items(), key=lambda kv: (kv[1], kv[0]))
    print([j[0] for j in ans][::-1])


arr = [
    1, 1, 1, 2, 3, 4, 9, 0, 2, 2, 3, 4, 3, 2, 1, 5, 5, 9, 0, 0, 1, 1, 1, 1, 2,
    5, 6, 7, 7, 8, 9, 0, 0, 4, 4, 4, 6, 6, 6, 6, 6, 7, 7, 7
]
func(arr)